File: 82-remove-duplicates-from-sorted-list-ii/remove-duplicates-from-sorted-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode p = head;

        Map<Integer, Integer> map = new HashMap<>();

        while(p != null){
            map.put(p.val, map.getOrDefault(p.val, 0) + 1);
            p = p.next;
        }

        p = new ListNode(0, head);
        ListNode g = p;

        while(p.next != null){
            if(map.get(p.next.val) > 1){
                p.next = p.next.next;
            }else{
                p = p.next;
            }
        }

        return g.next;
    }
}

File: 82-remove-duplicates-from-sorted-list-ii/remove-duplicates-from-sorted-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:

            arr = []

            curr = head

            dummy = ListNode(0)
            dum = dummy

            while curr:
                arr.append(curr.val)
                curr = curr.next
            
            ans = Counter(arr)

            for key,val in ans.items():

                if val == 1:

                    new = ListNode(key)

                    dum.next = new

                    dum = dum.next
            
            return dummy.next


            
        